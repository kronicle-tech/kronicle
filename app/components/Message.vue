<template>
  <b-alert v-if="message" show dismissible :variant="message.variant">
    <Markdown :markdown="message.markdown" />
  </b-alert>
</template>

<script lang="ts">
import Vue from 'vue'
import { BAlert } from 'bootstrap-vue'
import Markdown from '~/components/Markdown.vue'

interface Message {
  markdown: string
  variant?: string
}

export default Vue.extend({
  components: {
    'b-alert': BAlert,
    Markdown,
  },
  data() {
    return {
      message: this.$config.message as Message | undefined,
    }
  },
})
</script>
