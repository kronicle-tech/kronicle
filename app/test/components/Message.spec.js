import { mount } from '@vue/test-utils'
import Message from '@/components/Message.vue'

describe('Message', () => {
  let config
  let wrapper
  const createWrapper = () => {
    wrapper = mount(Message, {
      mocks: {
        $config: config,
      },
    })
  }

  beforeEach(() => {
    config = {}
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when messageMarkdown config is not set', () => {
    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual('')
    })
  })

  describe('when messageMarkdown config is set', () => {
    beforeEach(() => {
      config.messageMarkdown = '# Test Message'
    })

    test('renders an info alert showing the message as markdown', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
      expect(wrapper.html()).toEqual(
        expect.stringContaining('<h1>Test Message</h1>')
      )
      expect(wrapper.get('div[role="alert"]').classes()).toContain('alert-info')
    })

    describe('when messageVariant config is set', () => {
      beforeEach(() => {
        config.messageVariant = 'warning'
      })

      test('renders the alert with the specified variant', () => {
        createWrapper()
        expect(wrapper.html()).toMatchSnapshot()
        expect(wrapper.get('div[role="alert"]').classes()).toContain(
          'alert-warning'
        )
      })
    })
  })
})
