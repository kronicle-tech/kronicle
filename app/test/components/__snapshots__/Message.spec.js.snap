// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message when messageMarkdown config is set renders an info alert showing the message as markdown 1`] = `
"<transition-stub name=\\"\\" css=\\"true\\" enterclass=\\"\\" leaveclass=\\"show\\" entertoclass=\\"show\\" leavetoclass=\\"\\" enteractiveclass=\\"\\" leaveactiveclass=\\"\\">
  <div role=\\"alert\\" aria-live=\\"polite\\" aria-atomic=\\"true\\" class=\\"alert alert-dismissible alert-info\\"><button type=\\"button\\" aria-label=\\"Close\\" class=\\"close\\">×</button>
    <div>
      <h1>Test Message</h1>
    </div>
  </div>
</transition-stub>"
`;

exports[`Message when messageMarkdown config is set when message.variant config is set renders the alert with the specified variant 1`] = `
"<transition-stub name=\\"\\" css=\\"true\\" enterclass=\\"\\" leaveclass=\\"show\\" entertoclass=\\"show\\" leavetoclass=\\"\\" enteractiveclass=\\"\\" leaveactiveclass=\\"\\">
  <div role=\\"alert\\" aria-live=\\"polite\\" aria-atomic=\\"true\\" class=\\"alert alert-dismissible alert-warning\\"><button type=\\"button\\" aria-label=\\"Close\\" class=\\"close\\">×</button>
    <div>
      <h1>Test Message</h1>
    </div>
  </div>
</transition-stub>"
`;
